library/system/io.lean

```diff
@@ -68,6 +91,14 @@ interface.iterate e α a f
 def forever {e} (a : io_core e unit) : io_core e unit :=
 iterate () $ λ _, a >> return (some ())
 
+def catch {e₁ e₂ α} (a : io_core e₁ α) (b : e₁ → io_core e₂ α) : io_core e₂ α :=
+interface.catch e₁ e₂ α a b
+
+instance : alternative io :=
+{ interface.monad _ with
+  orelse := λ _ a b, catch a (λ _, b),
+  failure := λ _, io.fail "failure" }
+
```

Comment:
> Having said that, at the moment alternative extends applicative and not monad, I think we need to change that since we can't get the laws for bind from the laws for seq. 

Yes, `alternative` should inherit from `monad`, or we should add an `alternative_monad` type class.